src/Microsoft.OData.Core/UriParser/Binders/RangeVariableBinder.cs

```diff
@@ -24,6 +25,16 @@ internal static SingleValueNode BindRangeVariableToken(RangeVariableToken rangeV
         {
             ExceptionUtils.CheckArgumentNotNull(rangeVariableToken, "rangeVariableToken");
 
+            // If $this is the rangeVariableToken Name, we push a rangeVariable with the same TypeReference as the ImplicitRangevariable
+            // to the state.RangeVariables stack.
+            if (rangeVariableToken.Name == ExpressionConstants.This)

```

Comment:
>rangeVariableToken.Name == ExpressionConstants.This) [](start = 16, length = 52)

we don't have the logic for $it?

Reply:
For explicit $it, we create the RangeVariable in the `SelectExpandBinder`. It doesn't matter if we have $it the query string or not. For $this, I thought we should just create a rangeVariable whenever we have a $this rangeVariableToken. 
We can however redesign it to work similar with $it. https://github.com/OData/odata.net/blob/3f62d72c9d909dde519f130ffd8411f31ee95ca0/src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs#L894

Reply:
Maybe you could explain why they're or should be handled differently?

Reply:
@habbes My thinking was we were not to add a `$this` in the rangeVariables stack unless there is a `$this` literal in the query string. However to avoid confusion, I have moved the code to the SelectExpandBinder to work with the same logic as the `$it` rangeVariable. There are no performance improvements on adding the rangeVariable in the rangeVariable binder.

Reply:
#Resolved